src/dns/client.c

```diff
@@ -749,8 +749,10 @@ int dnsc_query(struct dns_query **qp, struct dnsc *dnsc, const char *name,
 	       uint16_t type, uint16_t dnsclass,
 	       bool rd, dns_query_h *qh, void *arg)
 {
-	if (!dnsc)
+	if (!dnsc) {
+		DEBUG_WARNING("DNS query attempted but DNS client is unconfigured\n");
 		return EINVAL;
+	}
```

Comment:
most of the code in re does not print a warning if a required parameter is NULL.

I think it would be better to move this check/warning to the SIP stack and check for `dnsc`
explicitly there.